src/components/icon/icon.pw.tsx

```diff
@@ -0,0 +1,548 @@
+import React from "react";
+import { test, expect } from "@playwright/experimental-ct-react17";
+import { IconProps } from "components/icon";
+import { IconComponent, IconTooltipComponent } from "./component.test-pw";
+import { getDataElementByValue } from "../../../playwright/components/index";
+import { SIZE, COLOR, CHARACTERS } from "../../../playwright/support/constants";
+
+import { checkAccessibility } from "../../../playwright/support/helper";
+
+const testData = [CHARACTERS.DIACRITICS, CHARACTERS.SPECIALCHARACTERS];
+const colorData = [COLOR.ORANGE, COLOR.RED, COLOR.BLACK, COLOR.BROWN];
+
+test.describe("should check Icon component properties", () => {
+  [true, false].forEach((boolVal) => {
+    test(`should check when aria-hidden prop is set as ${boolVal}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<IconComponent aria-hidden={boolVal} />);
+
+      const iconLocator = page.getByTestId("icon");
+
+      if (boolVal === true) {
+        await expect(iconLocator).toHaveAttribute("aria-hidden", "true");
+      } else {
+        await expect(iconLocator).toHaveAttribute("aria-hidden", "false");
+      }
```

Comment:
This if statement could be replaced with: 
`await expect(iconLocator).toHaveAttribute("aria-hidden", boolVal.toString());`